src/test/functional/functional-tests.js

```diff
@@ -872,8 +899,55 @@ describe('functional tests', function() {
         .catch(() => done())
     })
 
-    step('removeObject(bucketName, objectName, done)__', done => {
-      client.removeObject(bucketName, _1byteObjectName, done)
+    step('generatePresignedUrl(listObjectMethod, bucketName, \'\', expires, reqParams, cb)__', done => {
+      client.generatePresignedUrl('GET', bucketName, '', 1000, {'prefix': 'data', 'max-keys': 1000}, (e, presignedUrl) => {
+        if (e) return done(e)
+        var transport = http
+        var options = _.pick(url.parse(presignedUrl), ['hostname', 'port', 'path', 'protocol'])
+        options.method = 'GET'
+        options.headers = {
+        }
+        var str = ''
+        if (options.protocol === 'https:') transport = https
+        callback = function (response) {
+          if (!/20\d/.test(response.statusCode)) return done(new Error(`error on put : ${response.statusCode}`))
```

Comment:
GET will always have 200 for success so you dont have to worry about handling 204

Reply:
Done.